test/pub_get_and_upgrade_test.dart

```diff
@@ -34,57 +34,21 @@ main() {
       });
     });
 
-    test('adds itself to the packages directory and .packages file', () async {
+    test('adds itself to the .packages file', () async {
       // The package should use the name in the pubspec, not the name of the
       // directory.
       await d.dir(appPath, [
         d.pubspec({"name": "myapp_name"}),
         d.libDir('myapp_name')
       ]).create();
 
-      await pubCommand(command, args: ["--packages-dir"]);
-
-      await d.dir(packagesPath, [
-        d.dir("myapp_name",
-            [d.file('myapp_name.dart', 'main() => "myapp_name";')])
-      ]).validate();
+      await pubCommand(command);
 
       await d.dir("myapp", [
         d.packagesFile({"myapp_name": "."})
       ]).validate();
     });
 
-    test(
```

Comment:
Should these be checking the package spec? Or does it just not really matter?

Reply:
The current behavior is to include the package in the package spec regardless of whether is has a lib directory. I'm assuming that behavior is OK...